bin/concatenate_all.py

```diff
@@ -138,145 +205,80 @@ def main():
+    # optional field -> other_gradeable_paths=None if key doesn't exist
+    other_gradeable_paths = config.get("other_gradeable_paths")
 
     # ==========================================================================
     # loop through and concatenate the selected files for each user in this gradeable
     total_concat = 0
 
     for dir in regex_dirs:
-        gradeable_path = os.path.join(args.datapath, semester, course, dir, gradeable)
-        # loop over each user
-        for user in sorted(os.listdir(gradeable_path)):
-            user_path = os.path.join(gradeable_path, user)
-            if not os.path.isdir(user_path):
-                continue
-            elif user in users_to_ignore:
-                continue
-
-            if version_mode == "active_version":
-                # get the user's active version from their settings file if it exists, else get
-                # most recent version for compatibility with early versions of Submitty
-                submissions_details_path = os.path.join(user_path, 'user_assignment_settings.json')
-                if os.path.exists(submissions_details_path):
-                    with open(submissions_details_path) as details_file:
-                        details_json = json.load(details_file)
-                        my_active_version = int(details_json["active_version"])
-                else:
-                    # get the most recent version
-                    my_active_version = sorted(os.listdir(user_path))[-1]
-
-            # loop over each version
-            for version in sorted(os.listdir(user_path)):
-                version_path = os.path.join(user_path, version)
-                if dir == "results":
-                    # only the "details" folder within "results" contains files relevant to Lichen
-                    version_path = os.path.join(version_path, "details")
-                if not os.path.isdir(version_path):
-                    continue
-                if version_mode == "active_version" and int(version) != my_active_version:
-                    continue
-
-                output_file_path = os.path.join(args.basepath, "users", user,
-                                                version, "submission.concatenated")
-
-                if not os.path.exists(os.path.dirname(output_file_path)):
-                    os.makedirs(os.path.dirname(output_file_path))
-
-                # append to concatenated file
-                with open(output_file_path, "a") as output_file:
-                    concatenated_contents = getConcatFilesInDir(version_path, regex_patterns)
-                    output_file.write(concatenated_contents)
-                    total_concat += sys.getsizeof(concatenated_contents)
-
-                checkTotalSize(total_concat)
+        input_path = os.path.join(args.datapath, semester, course, dir, gradeable)
+        output_path = os.path.join(args.basepath, "users")
+        total_concat = processGradeable(args.basepath, config,
```

Comment:
The point of this variable was to check that cumulatively, we are not going over the size limit for the overall files we are concatenating. I think you forgot the `+`:
```suggestion
        total_concat += processGradeable(args.basepath, config,
```

Reply:
Hmm.  It's a little confusing perhaps.  The + is missing intentionally.  `total_concat` is passed to `processGradeable()` which increments it a number of times and returns the sum.  The reason we want this is because we want to check if we've exceeded the limit in the middle of `processGradeable()`, instead of just at the end.  Thoughts on how to do it better?